contracts/RandomGames.sol

```diff
@@ -0,0 +1,78 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;
+
+import {ERC721CM, ERC721ACQueryable, IERC721A} from "./ERC721CM.sol";
+
+/**
+ * @title RandomGamesMinting
+ */
+contract RandomGamesMinting is ERC721CM {
+    address private _proxyContract;
+
+    constructor(
+        string memory collectionName,
+        string memory collectionSymbol,
+        string memory tokenURISuffix,
+        uint256 maxMintableSupply,
+        uint256 globalWalletLimit,
+        address cosigner,
+        uint64 timestampExpirySeconds,
+        address mintCurrency,
+        address fundReceiver,
+        address proxyContract
+    )
+        ERC721CM(
+            collectionName,
+            collectionSymbol,
+            tokenURISuffix,
+            maxMintableSupply,
+            globalWalletLimit,
+            cosigner,
+            timestampExpirySeconds,
+            mintCurrency,
+            fundReceiver
+        )
+    {
+        _proxyContract = proxyContract;
```

Comment:
`require(newProxyContract != address(0), "Proxy contract cannot be zero address");`

You may want to deploy without setting an address, I see your setter function. Then I would add a similar check to the `mint` and `mintWithLimit` functions to ensure the `_proxyContract` address has been set at the beginning of the funciton.